nest-tax-backend/src/tax/utils/unified-tax.util.ts

```diff
@@ -0,0 +1,406 @@
+              .tz(`${taxYear}-${thirdPaymentDueDate}`, bratislavaTimeZone)
+              .toDate(),
+            status: installmentAmounts[2].status,
+            remainingAmount: installmentAmounts[2].toPay,
+          },
+        ]
+      : [
+          {
+            installmentNumber: 1,
+            dueDate: dueDate?.toDate(),
+            status: InstallmentPaidStatusEnum.AFTER_DUE_DATE,
+            remainingAmount: 0,
+          },
+          {
+            installmentNumber: 2,
+            dueDate: dayjs
+              .tz(`${taxYear}-${secondPaymentDueDate}`, bratislavaTimeZone)
+              .toDate(),
+            status: installmentAmounts[0].status,
+            remainingAmount:
+              installmentAmounts[1].toPay + installmentAmounts[0].toPay,
+          },
+          {
+            installmentNumber: 3,
+            dueDate: dayjs
+              .tz(`${taxYear}-${thirdPaymentDueDate}`, bratislavaTimeZone)
+              .toDate(),
+            status: installmentAmounts[2].status,
+            remainingAmount: installmentAmounts[2].toPay,
+          },
+        ]
+
+  const active = installmentDetails.find(
+    (installment) =>
+      installment.status === InstallmentPaidStatusEnum.NOT_PAID ||
+      installment.status === InstallmentPaidStatusEnum.PARTIALLY_PAID,
+  )
+
+  if (!active) {
+    return {
+      isPossible: false,
+      reasonNotPossible: InstallmentPaymentReasonNotPossibleEnum.ALREADY_PAID,
+    }
+  }
+
+  let paymentNote
+  if (active.installmentNumber === 1) {
+    paymentNote = QrPaymentNoteEnum.QR_firstInstallment
+  } else if (active.installmentNumber === 2) {
+    paymentNote =
+      installmentDetails[0].status === InstallmentPaidStatusEnum.AFTER_DUE_DATE
+        ? QrPaymentNoteEnum.QR_firstSecondInstallment
+        : QrPaymentNoteEnum.QR_secondInstallment
+  } else {
+    paymentNote = QrPaymentNoteEnum.QR_thirdInstallment
+  }
+
+  const activeInstallment = {
+    remainingAmount: active?.remainingAmount,
+    variableSymbol,
+    qrCode: {
+      amount: active.remainingAmount,
+      variableSymbol,
+      specificSymbol: '2025200000',
+      paymentNote,
+    },
+  }
+
+  return {
+    isPossible: true,
+    installments: installmentDetails,
+    activeInstallment,
+  }
+}
+
+const calculateOneTimePaymentDetails = (
+  overallPaid: number,
+  overallBalance: number,
+  dueDate: Date | undefined,
+  variableSymbol: string,
+): ReplaceQrCodeWithGeneratorDto<ResponseOneTimePaymentDetailsDto> => {
+  if (overallBalance === 0) {
+    return {
+      isPossible: false,
+      reasonNotPossible: OneTimePaymentReasonNotPossibleEnum.ALREADY_PAID,
+    }
+  }
+  return {
+    isPossible: true,
+    type:
+      overallPaid > 0
+        ? OneTimePaymentTypeEnum.REMAINING_AMOUNT_PAYMENT
+        : OneTimePaymentTypeEnum.ONE_TIME_PAYMENT,
+    amount: overallBalance,
+    dueDate,
+    qrCode: {
+      amount: overallBalance,
+      variableSymbol,
+      specificSymbol: '2025200000',
```

Comment:
This is not the place to hard code the value, if you want to hard code it please at least do it in the `TaxService` as an constant there. I also see it's duplicated here multiple times.